### PR TITLE
download sources via git for XGBoost 20171120 easyconfig due to use of submodules

### PR DESCRIPTION
**Pull request by mboisson**
This is an example of recipe that uses feature included in ~~https://github.com/easybuilders/easybuild-framework/pull/2555~~

### CODE REVIEW

**Comment by akesandgren** [comment]
@mboisson Add checksums please

**Comment by akesandgren** [comment]
Ah, didn't notice the recursive...

Kicking Travis due to changes in EasyBuild testing

**Comment by akesandgren** [comment]
Travis kicked

**Comment by akesandgren** [comment]
Some local variable name changes needed.

File: easybuild/easyconfigs/x/XGBoost/XGBoost-20171120-intel-2017b-Java-1.8.0_152.eb [hunk 1]
```diff
@@ -11,12 +11,15 @@ description = """XGBoost is an optimized distributed gradient boosting library d
...
+	'git_config': {
+		'url': 'https://github.com/dmlc',
+		'repo_name': 'xgboost',
+		'commit': local_commit,
+		'recursive': True,
+	},
+}]
 
 builddependencies = [('CMake', '3.9.5')]
 dependencies = [('Java', '1.8.0_152', '', True)]
```
Comment (akesandgren) [inline easybuild/easyconfigs/x/XGBoost/XGBoost-20171120-intel-2017b-Java-1.8.0_152.eb]: Better to use '1.8' here (didn't see that one before)

**Review by akesandgren — DISMISSED** [review]

File: easybuild/easyconfigs/x/XGBoost/XGBoost-20171120-intel-2017b-Java-1.8.0_152.eb (hunk 1 above)
Comment (akesandgren) [reply easybuild/easyconfigs/x/XGBoost/XGBoost-20171120-intel-2017b-Java-1.8.0_152.eb]: And that would mean changing the name of the easyconfig to match.

**Comment by akesandgren** [comment]
Also there is a conflict of some kind to resolve.

**Comment by mboisson** [comment]
Mmm, there should be no more conflict. Closing and reopening

**Comment by akesandgren** [comment]
Travis still thinks there is a conflict, can't see what it is...

**Comment by boegel** [comment]
@mboisson Please merge https://github.com/ComputeCanada/easybuild-easyconfigs/pull/16, which should make this PR (finally) too to go...

**Comment by boegel** [comment]
Test report by @boegel
**FAILED**
Build succeeded for 0 out of 1 (1 easyconfigs in total)
node3129.skitty.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) Gold 6140 CPU @ 2.30GHz, Python 3.6.8
See https://gist.github.com/9b505fb0043b3c2fd41c6abb26fb6ad6 for a full test report.

**Comment by boegel** [comment]
Hmm, tests seem to be failing due to `java.net.ConnectException: Connection refused (Connection refused)`

**Comment by boegel** [comment]
> Hmm, tests seem to be failing due to `java.net.ConnectException: Connection refused (Connection refused)`

Looks like a known issue with the tests failing in some network configurations, see https://github.com/dmlc/xgboost/issues/4484...

My vote would go to disabling that test.

**Comment by boegel** [comment]
Looks like the failing tests were a fluke, they're passing now (see upcoming test report), so this should be good to go...

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in total)
node2617.swalot.os - Linux centos linux 7.9.2009, x86_64, Intel(R) Xeon(R) CPU E5-2660 v3 @ 2.60GHz (haswell), Python 3.6.8
See https://gist.github.com/d3597c7ba7331a10ab32e1e5af1db541 for a full test report.

**Review by boegel — APPROVED** [review]
lgtm

**Comment by boegel** [comment]
Going in, thanks @mboisson!

**Pull request merged**